fluss-server/src/main/java/com/alibaba/fluss/server/kv/rocksdb/RocksDBMetricsManager.java

```diff
@@ -0,0 +1,223 @@
+            LOG.warn("Cannot register collector after collector is shutdown");
+            return;
+        }
+
+        String key = createCollectorKey(collector);
+        collectors.put(key, collector);
+
+        // Start the collection if this is the first collector
+        if (collectors.size() == 1 && !running.get()) {
+            startCollection();
+        }
+
+        LOG.debug(
+                "Registered RocksDB metrics collector for {}, total collectors: {}",
+                key,
+                collectors.size());
+    }
+
+    /**
+     * Unregister a metrics collector.
+     *
+     * @param collector the collector to unregister
+     */
+    public void unregisterCollector(RocksDBMetricsCollector collector) {
+        String key = createCollectorKey(collector);
+        collectors.remove(key);
+
+        LOG.debug(
+                "Unregistered RocksDB metrics collector for {}, remaining collectors: {}",
+                key,
+                collectors.size());
+    }
+
+    /** Start the metrics collection if not already running. */
+    private void startCollection() {
+        if (running.compareAndSet(false, true)) {
+            LOG.info(
+                    "Starting RocksDB metrics collection with {} threads",
+                    METRICS_COLLECTION_THREADS);
+
+            metricsCollectionExecutor.scheduleAtFixedRate(
+                    this::gatherMetrics,
+                    METRICS_UPDATE_INTERVAL_SECONDS,
+                    METRICS_UPDATE_INTERVAL_SECONDS,
+                    TimeUnit.SECONDS);
+        }
+    }
+
+    /** Collect metrics from all registered collectors in batches. */
+    private void gatherMetrics() {
+        if (collectors.isEmpty()) {
+            return;
+        }
+
+        List<RocksDBMetricsCollector> collectorList = new ArrayList<>(collectors.values());
+
+        // Process collectors in batches to avoid overwhelming the system
+        for (int i = 0; i < collectorList.size(); i += BATCH_SIZE) {
+            int endIndex = Math.min(i + BATCH_SIZE, collectorList.size());
+            List<RocksDBMetricsCollector> batch = collectorList.subList(i, endIndex);
+
+            // Submit batch for processing
+            metricsCollectionExecutor.submit(() -> processBatch(batch));
+        }
+    }
+
+    /** Process a batch of collectors. */
+    private void processBatch(List<RocksDBMetricsCollector> batch) {
+        for (RocksDBMetricsCollector collector : batch) {
+            try {
+                collector.updateMetrics();
+            } catch (Exception e) {
+                LOG.warn(
+                        "Error updating metrics for collector {}",
+                        createCollectorKey(collector),
+                        e);
+            }
+        }
+    }
+
+    /** Create a unique key for a collector. */
+    private String createCollectorKey(RocksDBMetricsCollector collector) {
+        return String.format(
+                "%s-%s-%s-%s",
```

Comment:
`-` is a valid character in database name and table name, therefore, it's better to use `/` as separator, to avoid the key conflicts.